### PR TITLE
Add Ctrl+F shortcut for the search bar

### LINKED ISSUE #18897: Catch CTRL+F and focus search input
For better UX we could react on CTRL+F and focus the search input if available.

@jancborchardt @PVince81 @MorrisJobke

### PR DESCRIPTION
**Pull request by Henni**
fixes #18897 

cc @ChristophWurst @jancborchardt @PVince81 @MorrisJobke

### CODE REVIEW

**Comment by jancborchardt** [comment]
:+1: cool, thank you! :)

**Comment by ChristophWurst** [comment]
@Henni I get JS errors:
```
ReferenceError: e is not defined                       search.js:347:0
```
Tested with FF 40.0.3 on arch linux

**Comment by Henni** [comment]
@ChristophWurst fixed

**Comment by nickvergessen** [comment]
Do we really want to override a globally known universal browser shortcut?
Preventing full page search and replacing it with inaccurate search results sounds not very useful to me.

**Comment by Henni** [comment]
@nickvergessen take a look at #18897. We already discussed about that over there.

**Comment by jancborchardt** [comment]
@nickvergessen please read the reasoning in https://github.com/owncloud/core/issues/18897

**Comment by nickvergessen** [comment]
@MorrisJobke also disagreed in the issue

**Comment by ChristophWurst** [comment]
@Henni works now, thanks :+1:

File: core/search/js/search.js [hunk 1]
```diff
@@ -342,6 +342,13 @@
 					self.hideResults();
 				}
 			});
+
```
Comment (ChristophWurst) [inline core/search/js/search.js]: this line should be empty

Comment (Henni) [inline core/search/js/search.js]: @ChristophWurst thanks. Fixed!

**Comment by jancborchardt** [comment]
@nickvergessen we put some thought in it, read replies in the issue.

**Comment by jancborchardt** [comment]
This is awesome! :+1: on the Apps management page I always find myself using Ctrl+F out of habit, but using the built-in search is much better.

**Comment by schiesbn** [comment]
I agree with @MorrisJobke and @nickvergessen. I don't think it is a good idea to change well known default-shortcuts of the browser. People expect that Ctrl-F works on ownCloud like on any other web page. Also it doesn't always make sense to use the build-in search. Sometimes you want to search stuff which isn't indexed by our search but still displayed on the page. Why don't just define our own shortcut for the ownCloud-search?

**Comment by jancborchardt** [comment]
Sorry but this argument is moot. As comparison: When using an online text editor (or other app with a save function), then  Ctrl+S should also not bring up the »Save website« dialog. Sure people might »expect it«, but it makes no sense whatsoever to bring up that dialog instead of just saving that document. Oh right, we already do it exactly like that (hijacking Ctrl+S) in our text editor – works great.

There’s a difference between web **page** and web **app**.

**Comment by nickvergessen** [comment]
Ctrl+S is a completly different story. It is of no real use when inside the text editor, agreed.

But Ctrl+F has a use case outside of "I want to use the search of ownCloud" when being pressed,
which you don't use but others do use.

I don't really get your page vs. app thing, we display text that people want to search through, we don't only have web app content.

**Comment by jancborchardt** [comment]
> But Ctrl+F has a use case outside of "I want to use the search of ownCloud" when being pressed,
which you don't use but others do use.

Then please be clear about the cases. The built-in search is much better for the files app instead of Ctrl+F.

**Comment by nickvergessen** [comment]
E.g. I do Ctrl+F on the app list to jump to the experimental section, or to find apps by author names etc.

**Comment by jancborchardt** [comment]
> E.g. I do Ctrl+F on the app list to jump to the experimental section, or to find apps by author names etc.

**Comment by jancborchardt** [comment]
> E.g. I do Ctrl+F on the app list to jump to the experimental section, or to find apps by author names etc.

Simple. Typing »Exp…« should bring up the experimental section. And of course the search should go through every attribute like author names, description etc. @icewind1991

**Comment by LukasReschke** [comment]
Agreeing with @nickvergessen FWIW.

**Comment by jancborchardt** [comment]
Or it would be a way to kick the asses of these »power users« (of which some are ownCloud devs) to then just go ahead and implement it to scratch their own itch. But if we don’t start with one point then progress will stall.

**Comment by nickvergessen** [comment]
> Or it would be a way to kick the asses of these »power users« (of which some are ownCloud devs) to then just go ahead and implement it to scratch their own itch. But if we don’t start with one point then progress will stall.

Ever looked at our open issues and features we are implementing? 

aint-nobody-got-time-for-that

**Comment by jancborchardt** [comment]
@nickvergessen please stop the whining. This is slated for 9.0, and if you think that search in app mgmt could use some improvements -> FIX IT! :)

**Comment by nickvergessen** [comment]
Or you stop killing browser features and replacing them with non-existing owncloud features?
I'm out of this

**Comment by MorrisJobke** [comment]
https://github.com/owncloud/core/issues/18897#issuecomment-142573463

**Comment by MorrisJobke** [comment]
I also don't like this yet. Once it works properly in ownCloud we can enable it but not the other way around.

@jancborchardt And shouting to everybody "Fix it" doesn't fix problems. Maybe you should also fix some stuff ;) -> lets move this once 8.2 is out.

**Comment by jancborchardt** [comment]
@Henni can you rebase this? :)

@PVince81 @DeepDiver1975 what do you think here? I’d like to move forward and not sink into ideological discussions that web apps shouldn’t hijack any shortcuts (which we do already). The search is difficult to reach as it is, and this shortcut would aid.

**Comment by Henni** [comment]
I created a separate PR which uses the shortcut Ctrl+Shift+F (#20475).
Maybe this helps decide which of these approaches is better suited for owncloud.

**Comment by PVince81** [comment]
@Henni thanks. Would be good to let many people try this out before making a decision.

I'd rather tend to go the route "if it ain't broke, don't fix it" and do nothing.

**Comment by jancborchardt** [comment]
@Henni I’d say we should definitely catch Ctrl-F since that’s the usual one. Ctrl-Shift-F we can do in addition to be forgiving.

@PVince81 I think most concerns in this issue are based on the theoretical assumption that »we shouldn’t do it because reasons«. This is not how we move forward, and how we make the search functionality useful and quick to use.

**Comment by jancborchardt** [comment]
Please review @owncloud/designers

**Comment by Henni** [comment]
I added a fallback so that pressing Ctrl+F, when the searchbox is already focused, the browser default behavior is executed.

@jancborchardt @MorrisJobke @nickvergessen @schiesbn @LukasReschke @PVince81 are you guys happy with this fallback?

File: core/search/js/search.js [hunk 2]
```diff
@@ -19,6 +19,7 @@
 	 * @param $searchResults container element for results und status message
 	 */
 	var Search = function($searchBox, $searchResults) {
+		console.log('init');
```
Comment (nickvergessen) [inline core/search/js/search.js]: wops

**Comment by nickvergessen** [comment]
Sounds good to me, confused people looking for the browser search will most likely hit ctrl+f again and then end up where they want to be

File: core/search/js/search.js (hunk 2 above)
Comment (Henni) [inline core/search/js/search.js]: jep sry

**Comment by jancborchardt** [comment]
Ok, let’s do this :+1: 

@nickvergessen can you test and review so we can get this in?

**Comment by nickvergessen** [comment]
:+1:

**Comment by jancborchardt** [comment]
@DeepDiver1975 merge please :)

**Comment by jancborchardt** [comment]
Nice, finally! This makes search so much easier!

mimimi don’t overwrite browser shortcuts

**Pull request merged**